static/app/components/globalDrawer/components.tsx

```diff
@@ -0,0 +1,96 @@
+import {useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {Button} from 'sentry/components/button';
+import type {DrawerOptions} from 'sentry/components/globalDrawer';
+import {IconClose} from 'sentry/icons/iconClose';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import useKeyPress from 'sentry/utils/useKeyPress';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+import SlideOverPanel from 'sentry/views/insights/common/components/slideOverPanel';
+
+export interface DrawerPanelProps extends DrawerOptions {
```

Comment:
props like onClose should always be defined, i think the props should be something like `Required<DrawerOptions>` or something that is a bit closer to how its being used